### ISSUE TITLE
Yaml Secret and FeatureFlag deployment via App Configuration Task

### ISSUE THREAD

**Issue by JDinisHiscox** [Azure/AppConfiguration#732]
Please add support deploy Yaml secrets and featureflags with the App Configuration Task.

At the moment we can deploy an yaml with the format:
```YAML
Test:
   uri : >-
     "https://....vault.azure.net/secretname"
```

This gets transformed into the following json **if the content-type is not** `application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8` or `application/vnd.microsoft.appconfig.ff+json;charset=utf-8`:

```json
{ "url": "https://....vault.azure.net/secretname" }
```

So here are my two suggestions:

1 -  Accept an yaml file for `application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8` and `application/vnd.microsoft.appconfig.ff+json;charset=utf-8` content-types

or

2 - Add support to import yaml with content-types: `application/vnd.microsoft.appconfig.keyvaultref+yaml;charset=utf-8` or `application/vnd.microsoft.appconfig.ff+yaml;charset=utf-8`

**Comment by MaryanneNjeri** [comment]
@JDinisHiscox for Yaml feature flag deployment via the Azure App Configuration PushTask, you can specify a feature management section, please see example in  [Azure App Configuration support for configuration file](https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-config-file#file-content-profile-default). 

For the key vault references, sounds valid to support deploying secrets with a YAML file. I can look into this and get back to you

**Comment by JDinisHiscox** [comment]
For features im trying to import unsuccessfully via the Azure App Configuration Push Task the following yamls:

Feature Flags:
```yaml
FeatureManagement:
  "flags.eu.appname.operations.operationname": true
```

Secrets:
```yaml
variables:
  "eu:appname:backend:secret:value":
    uri: "secreturi"
```

Feature Flags keys cant also contain : but i guess i can live fine with the .

Via cli these get imported without any issue however thru the task they fail because of the content-type because it seems to attempt to parse the contents of the file based on the content-type passed.

Via portal only secrets gets loaded, feature flags either dont show up or show up but not in the feature management section.

Thanks for looking into this 👍😄

**Comment by zhenlan** [comment]
Hi @JDinisHiscox I am wondering if you get a chance to check out this document.
https://learn.microsoft.com/en-us/azure/azure-app-configuration/concept-config-file

**Comment by JDinisHiscox** [comment]
Hi @zhenlan,

I did, have i missed something?

Going by what it says App Configuration supports JSON, Yaml, or Properties file formats, however when deploying secrets or flags im met with the following with yamls having the same format i've described above:

FeatureFlags:
```
Azure App Configuration Push Task is starting up...
Azure Subscription: ...
App Configuration Endpoint: https:/....azconfig.io
Strict: false
File Content Profile: appconfig/default
Dry Run: false
Import Mode: All
Separator: ':'
Label: "DEV"
##[error]An unexpected error occurred. Failed to import 'Yaml' file format with 'application/vnd.microsoft.appconfig.ff+json;charset=utf-8' content type. Please provide JSON file format to match your content type.
```

Secrets:
```
Azure App Configuration Push Task is starting up...
Azure Subscription: ...
App Configuration Endpoint: https://....azconfig.io/
Strict: false
File Content Profile: appconfig/default
Dry Run: false
Import Mode: All
Separator: ':'
Label: "DEV"
##[error]An unexpected error occurred. Failed to import 'Yaml' file format with 'application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8' content type. Please provide JSON file format to match your content type.
```

Here are the yaml pipeline tasks:

```yaml
  - task: AzureAppConfigurationPush@5
    displayName: Secrets Deployment
    inputs:
      azureSubscription: '${{parameters.azureServiceConnection}}'
      AppConfigurationEndpoint: '${{parameters.appConfigEndpoint}}'
      ConfigurationFile: '$(System.DefaultWorkingDirectory)/$(resources.pipeline.build.runName)/Resources/${{parameters.environment}}/secrets.yml'
      Separator: ':'
      ContentType: 'application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8'
      Label: 'DEV'
      Strict: false
      DryRun: false

  - task: AzureAppConfigurationPush@5
    displayName: Flags Deployment
    inputs:
      azureSubscription: '${{parameters.azureServiceConnection}}'
      AppConfigurationEndpoint: '${{parameters.appConfigEndpoint}}'
      ConfigurationFile: '$(System.DefaultWorkingDirectory)/$(resources.pipeline.build.runName)/Resources/${{parameters.environment}}/flags.yml'
      Separator: ':'
      Label: 'DEV'
      ContentType: 'application/vnd.microsoft.appconfig.ff+json;charset=utf-8'
      Strict: false
      DryRun: false
```

**Comment by zhenlan** [comment]
@JDinisHiscox, thanks for the confirmation. To help us repro the issue, can you please share a sample of your _secrets.yml_ and _flags.yml_?

Feature flags should be defined by following the format in yaml files below. The Push Task will add proper content type for feature flags automatically, so you shouldn't need to add it.
https://microsoft.github.io/spring-cloud-azure/docs/azure-app-configuration/2.9.0/reference/html/index.html#feature-flag-declaration.

**Comment by JDinisHiscox** [comment]
> For features im trying to import unsuccessfully via the Azure App Configuration Push Task the following yamls:
> 
> Feature Flags:
> 
> ```yaml
> FeatureManagement:
>   "flags.eu.appname.operations.operationname": true
> ```
> 
> Secrets:
> 
> ```yaml
> variables:
>   "eu:appname:backend:secret:value":
>     uri: "secreturi"
> ```
> 
> Feature Flags keys cant also contain : but i guess i can live fine with the .
> 
> Via cli these get imported without any issue however thru the task they fail because of the content-type because it seems to attempt to parse the contents of the file based on the content-type passed.
> 
> Via portal only secrets gets loaded, feature flags either dont show up or show up but not in the feature management section.
> 
> Thanks for looking into this 👍😄
> 
> P.S: Since i didnt make it clear, i'm trying to import settings, secrets and feature flags via an azure devops deployment pipeline with an yaml for each type of configuration value.
> 
> Trying to maintain the same aspect across configutation types and also here is an example for my settings.yml format:
> 
> ```yaml
> variables:
>   "eu:appname:backend:name": "backend name"
> ```

These are actually the contents from yamls we use stripped of private content.

**Comment by zhenlan** [comment]
> it seems to attempt to parse the contents of the file based on the content-type passed

@MaryanneNjeri, how do we tell if a file is json or yaml? File extension? It shouldn't be based on the content-type.

**Comment by zhenlan** [comment]
@MaryanneNjeri, are you able to repro the issue @JDinisHiscox reported? It looks like we mistreated the input file as json. We probably should ask for the file format explicitly like what we do in CLI.

**Comment by MaryanneNjeri** [comment]
@zhenlan yes am able to repro the on my end, we check if the content type is JSON and file format is not JSON we error out.

**Comment by zhenlan** [comment]
@MaryanneNjeri is this addressed?

**Comment by MaryanneNjeri** [comment]
@zhenlan yes this issue has been released in the latest Push task version 6.0.0
